Lib/tkinter/auto_complete.py

```diff
@@ -0,0 +1,155 @@
+try:
+    import tkinter as tk
+    from tkinter import ttk
+except ImportError:
+    # Python 2
+    import Tkinter as tk
+    import ttk
+
+__all__ = ["AutocompleteEntry"]
+
+NO_RESULTS_MESSAGE = "No results found for '{}'"
+
+
+class AutocompleteEntry(tk.Frame, object):
+    """A container for `tk.Entry` and `tk.Listbox` widgets.
+    An instance of AutocompleteEntry is actually a `tk.Frame`,
+    containing the `tk.Entry` and `tk.Listbox` widgets needed
+    to display autocompletion entries. Thus, you can initialize
+    it with the usual arguments to `tk.Frame`.
+    Constants:
+    LISTBOX_HEIGHT -- Default height for the `tk.Listbox` widget
+    LISTBOX_WIDTH -- Default width for the `tk.Listbox` widget
+    ENTRY_WIDTH -- Default width for the `tk.Entry` widget
+    Methods:
+    __init__ -- Set up the `tk.Listbox` and `tk.Entry` widgets
+    build -- Build a list of autocompletion entries
+    _update_autocomplete -- Internal method
+    _select_entry -- Internal method
+    _cycle_up -- Internal method
+    _cycle_down -- Internal method
+    Other attributes:
+    text -- StringVar object associated with the `tk.Entry` widget
+    entry -- The `tk.Entry` widget (access this directly if you
+             need to change styling)
+    listbox -- The `tk.Listbox` widget (access this directly if
+             you need to change styling)
+    """
+    LISTBOX_HEIGHT = 5
+    LISTBOX_WIDTH = 25
+    ENTRY_WIDTH = 25
+
+    def __init__(self, master, *args, **kwargs):
+        """Constructor.
+        Create the `self.entry` and `self.listbox` widgets.
+        Note that these widgets are not yet displayed and will only
+        be visible when you call `self.build`.
+        Arguments:
+        master -- The master tkinter widget
+        Returns:
+        None
+        """
+        super(AutocompleteEntry, self).__init__(*args, **kwargs)
+        self.text = tk.StringVar()
+        self.entry = tk.Entry(
+            self,
+            textvariable=self.text,
+            width=self.ENTRY_WIDTH
+        )
+        self.listbox = tk.Listbox(
+            self,
+            height=self.LISTBOX_HEIGHT,
+            width=self.LISTBOX_WIDTH
+        )
+
+    def build(
+              self,
+              entries,
```

Comment:
```suggestion
              *entries,
```
We could then accept values similar to an `OptionMenu`. tkinter already has the required flattening logic so this would not stop us accepting a list/tuple of entries.